### DISCUSSION TITLE
Cheaper N64 dev cart dumping solution

### DISCUSSION THREAD

**Discussion by mariomadproductions** [sanni/cartreader#295] [General Chat]
Hi @sanni, did you see @retrostage has implemented N64 dev cart support, maybe you can work together to bring that to this project too? https://twitter.com/RetroStage_PCB/status/1463375632012775425

Edit:
I see that sanni is looking for an affordable dev cart to test with - it says that on the wiki. And it seems like someone here is offering to lend(?) him one. I will also look out for anyone else who might be able to help with that, in some relevant communities.
https://github.com/sanni/cartreader/discussions/196#discussioncomment-1219473

**Comment by sanni** [comment]
If someone has both a dev cart and a Cart Reader (V3) he can try plugging in an 12V barrel plug power supply into the Arduino Mega barrel connector together with the standard 5V USB power supply and see if dumping works. Beforehand make sure using a multimeter that the 3.3V pins on the cart slot are 3.3V and the 12V pins on the cart slot are indeed 12V.

**Comment by ehw** [comment]
Hey sanni, this is ehw from Hidden Palace.

I was working with @retrostage to add support for the N64 flash dev carts over the past several months. From what was discovered, the N64 dev carts require additional voltage in order to be read which is why the newer retroblasters  will require two USB cables for power. I believe the carts require 12V even though the console outputs 3V, but I'm unsure.

I had a word with someone who can supply you a cart for testing. I'll put them in contact with you shortly. But afaik, that was the only change that was needed. It didn't need a CIC from a N64 that was mentioned elsewhere.

**Reply by mariomadproductions** [reply]
Nice work!

**Reply by jonasrosland** [reply]
This is great to see!

**Comment by sanni** [comment]
I'd rather wait for someone else to try it to be honest. 

Everything should be implemented both in hardware and in software to theoretically dump proto carts. You have to use the V3 Cart Reader built like shown in the build guide(it's important that the two parts were removed from the Arduino board as shown in Step 1) and then connect a 8-12V DC power supply to the barrel jack of the Arduino in addition to the USB one.

Just need to be super careful to not misalign the cartridge because the 12V pins are right next to some address pins so if the cartridge is inserted not perfectly centered it will be dead. Which is why I always tell people to use the UltraSave instead since it has a cart-sized shroud that ensures correct alignment.

**Reply by retrostage** [reply]
The connectors I use have the same pitch as an OEM connector, so alignment isn't an issue. If you're interested in some for your newest hardware revision let me know

**Reply by j4m13c0** [reply]
I will upgrade my current V2 to a V3 to try and dump some N64 Dev Carts. Do you think its possible to be able to write to the carts too? Is the protocol known?

**Reply by retrostage** [reply]
I would personally recommend against supporting writing to dev carts. That just opens Pandora's box, with a ton of cheap sports dev carts being turned into something much more valuable to try and fool collectors.

**Reply by j4m13c0** [reply]
> I would personally recommend against supporting writing to dev carts. That just opens Pandora's box, with a ton of cheap sports dev carts being turned into something much more valuable to try and fool collectors.

Yeah, that's true. That is the hard part about supporting the dev carts too. Its hard as a tinkerer I love to play with the dev hardware but at the same time, don't want to have people getting ripped off too.

**Reply by mariomadproductions** [reply]
An alternative might be to develop/look into a homebrew development cart. I.e. something that acts like a dev cart, but you can tell just from looking at the board/cart that it obviously isn't an official one.

**Reply by j4m13c0** [reply]
Yeah, that's true. I have a few N64 Dev carts and some of them are blank, I wanted to have a play about with them. Most of the other dev stuff I can use but the N64 ones are not so well documented.

**Reply by mariomadproductions** [reply]
Although on the other hand, getting blank dev carts, writing to them and then reading back the data and checking it matches would be a good way to test the dumping of different dev cart types.

**Reply by mariomadproductions** [reply]
Note about UltraSave: it supposedly produces bad dumps of NUS-8F16FB-02 development boards (see: https://dumping.guide/carts/nintendo/n64)

**Comment by Ancyker** [comment]
After talking to @PsyK0p4T in Discord I started looking at adding 12V to HW5
![](https://cdn.discordapp.com/attachments/1038265106179575911/1040477847908790362/Screenshot_921.png)
![](https://cdn.discordapp.com/attachments/1038265106179575911/1040477847594213386/Screenshot_920.png)

I'm not sure if CY201 is needed. Might just be able to bridge that. If someone can find a dev cart that isn't thousands of dollars I can get one to test it...

Edit: Just wanted to note that this was designed in a way that if it wasn't desired/needed you could just leave all the pads unpopulated.

**Comment by mariomadproductions** [comment]
@sanni It seems like Tony from Hard4Games and MarcoEagleEye managed to dump some N64 dev carts with the 12V line added. https://www.youtube.com/watch?v=OFmfTQez4Hg

I've asked if they can reply to this thread.

Although I think the safety features you mentioned should probably be included alongside this change, because we don't want people destroying their dev carts.
